src/Bicep.Core/Semantics/FileSymbol.cs

```diff
@@ -94,6 +111,46 @@ public override void Accept(SymbolVisitor visitor)
 
         public IEnumerable<DeclaredSymbol> GetDeclarationsByName(string name) => this.declarationsByName[name];
 
+        /// <summary>
+        /// Tries to get the semantic module of the Bicep File referenced via a using declaration from the current file.
+        /// If current file is not a parameter file, the method will return false.
+        /// </summary>
+        public bool TryGetBicepFileSemanticModelViaUsing([NotNullWhen(true)] out SemanticModel? semanticModel, [NotNullWhen(false)] out Diagnostic? failureDiagnostic)
```

Comment:
>TryGetBicepFileSemanticModelViaUsing [](http://example.com/codeflow?start=20&length=36)

`ModuleSymbol` has a similar method to get the semantic model of the referenced module or fail. Since there's no `UsingSymbol`, I put it on the file symbol.

Also open to less horrible naming :)

Reply:
Should we cache the lookup of `UsingDeclarationSyntax?` somewhere? Feels like this could be called fairly often - e.g. when calling `GetDeclaredParameterAssignmentType()`

Reply:
Yeah, we should.